debugger/src/cell_renderers/cellrendererbreakicon.c

```diff
@@ -200,33 +212,56 @@ static void cell_renderer_break_icon_get_size(GtkCellRenderer *cell, GtkWidget *
 /*
  * render a cell
  */
+#if GTK_CHECK_VERSION(3, 0, 0)
+static void cell_renderer_break_icon_render(GtkCellRenderer *cell, cairo_t *cr, GtkWidget *widget,
+	const GdkRectangle *background_area, const GdkRectangle *cell_area, GtkCellRendererState flags)
+#else
 static void cell_renderer_break_icon_render(GtkCellRenderer *cell, GdkDrawable *window, GtkWidget *widget,
 	GdkRectangle *background_area, GdkRectangle *cell_area, GdkRectangle *expose_area, GtkCellRendererState flags)
+#endif
 {
 	CellRendererBreakIcon *cellbreakpoint = (CellRendererBreakIcon*) cell;
 	
 	GdkPixbuf *pixbuf = NULL;
 	
 	GdkRectangle pix_rect;
 	GdkRectangle draw_rect;
+
+	gint xpad;
+	gint ypad;
+
+	GValue is_expander_value = G_VALUE_INIT;
+	gboolean is_expander;
+
+#if !GTK_CHECK_VERSION(3, 0, 0)
 	cairo_t *cr;
+#endif
 	
 	cell_renderer_break_icon_get_size (cell, widget, cell_area,
 		&pix_rect.x,
 		&pix_rect.y,
 		&pix_rect.width,
 		&pix_rect.height);
 	
-	pix_rect.x += cell_area->x + cell->xpad;
-	pix_rect.y += cell_area->y + cell->ypad;
-	pix_rect.width  -= cell->xpad * 2;
-	pix_rect.height -= cell->ypad * 2;
-	
+	gtk_cell_renderer_get_padding(cell, &xpad, &ypad);
+	pix_rect.x += cell_area->x + xpad;
+	pix_rect.y += cell_area->y + ypad;
+	pix_rect.width  -= xpad * 2;
+	pix_rect.height -= ypad * 2;
+
+#if GTK_CHECK_VERSION(3, 0, 0)
+	if (!gdk_rectangle_intersect (cell_area, &pix_rect, &draw_rect))
+#else
 	if (!gdk_rectangle_intersect (cell_area, &pix_rect, &draw_rect) ||
 		!gdk_rectangle_intersect (expose_area, &draw_rect, &draw_rect))
+#endif
 		return;
+
+	g_object_get_property(G_OBJECT(cell), "is-expander", &is_expander_value);
+	is_expander = g_value_get_boolean(&is_expander_value);
+	g_value_unset(&is_expander_value);
```

Comment:
You should probably just use the simpler version: `g_object_get(cell, "is-expander", &is_expander, NULL)`

Reply:
You're right, changed.